pkg/secrets-store/provider_client.go

```diff
@@ -263,19 +265,16 @@ func MountContent(ctx context.Context, client v1alpha1.CSIDriverProviderClient,
 		klog.InfoS("proto above 1MiB, secret sync may fail", "size", size)
 	}
 
-	if len(resp.GetFiles()) > 0 {
-		klog.V(5).Info("writing mount response files")
-		if err := fileutil.Validate(resp.GetFiles()); err != nil {
-			return nil, internalerrors.FileWriteError, err
-		}
-		if err := fileutil.WritePayloads(targetPath, resp.GetFiles()); err != nil {
-			return nil, internalerrors.FileWriteError, err
-		}
-	} else {
-		// when no files are returned we assume that the plugin has not migrated
-		// grpc responses for writing files yet.
-		klog.V(5).Info("mount response has no files")
+	if len(resp.GetFiles()) == 0 {
+		// The plugin mount response contains no files. Possible that the plugin
+		// is writing its own files instead of the driver (See Issue #551).
+		klog.V(5).Info("Empty files in mount response. It is possible that the plugin has not migrated to driver-written files (Issue #551).")
+	}
+
+	if err := fileutil.WritePayloads(targetPath, resp.GetFiles()); err != nil {
+		return nil, internalerrors.FileWriteError, err
 	}
+	klog.V(5).Info("mount response files written.")
```

Comment:
Added a comment on returning early above. If we don't return early, these message will conflict with [log](https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/1202/files#diff-39f63ba0bd3c5582f2919a5632b3b7d89d55a105ce5aecee4325fc146e8c5205R271) here causing confusion.

Reply:
:+1: